### PR TITLE
[body-parser] update to v1.16

### PR DESCRIPTION
**Pull request by tlaziuk**
- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/expressjs/body-parser/tree/1.16.0#body-parser>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

**Comment by aozgaa** [comment]
Thoughts @santialbo, @vilic, @dreampulse, @tlaziuk ?

**Comment by tlaziuk** [comment]
@aozgaa @santialbo @vilic @dreampulse
please take a look at the test file, is it a bug?

    body-parser-tests.ts(11,9): error TS2339: Property 'use' does not exist on type 'typeof e'.
    body-parser-tests.ts(12,9): error TS2339: Property 'use' does not exist on type 'typeof e'.
    body-parser-tests.ts(13,9): error TS2339: Property 'use' does not exist on type 'typeof e'.
    body-parser-tests.ts(14,9): error TS2339: Property 'use' does not exist on type 'typeof e'.
    body-parser-tests.ts(14,13): error TS2346: Supplied parameters do not match any signature of call target.
    body-parser-tests.ts(17,9): error TS2339: Property 'all' does not exist on type 'typeof e'.
    body-parser-tests.ts(17,19): error TS7006: Parameter 'req' implicitly has an 'any' type.
    body-parser-tests.ts(17,24): error TS7006: Parameter 'res' implicitly has an 'any' type.
    body-parser-tests.ts(17,29): error TS7006: Parameter 'next' implicitly has an 'any' type.
    body-parser-tests.ts(22,9): error TS2339: Property 'listen' does not exist on type 'typeof e'.

File: body-parser/body-parser-tests.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    json,
+    raw,
+    text,
+    urlencoded,
+} from 'body-parser';
+
+const app = express();
+
+express.use(json());
+express.use(raw());
```
Comment (minestarks) [inline body-parser/body-parser-tests.ts]: You're initializing the variable ```app``` but not actually using it. ```express``` is the constructor, not the instance, which is why you're getting errors.

**Review by minestarks — CHANGES_REQUESTED** [review]

**Comment by minestarks** [comment]
@tlaziuk Please fix the failing tests before I can merge. Thank you.

File: body-parser/body-parser-tests.ts (hunk 1 above)
Comment (tlaziuk) [inline body-parser/body-parser-tests.ts]: thanks, I didn't notice that before! 👍

**Review by minestarks — APPROVED** [review]

**Comment by jcristovao** [comment]
Hello,

I'm sorry, but this PR introduced a typo: [reviever](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/14493/files#diff-69c3b5d3084ae119dda4ece757760d85R25) instead of [reviver](https://github.com/expressjs/body-parser/blob/master/lib/types/json.js#L57).

@tlaziuk , do you think you can quickly fix this? Thanks

**Pull request merged**